airflow/jobs/scheduler_job.py

```diff
@@ -901,6 +901,33 @@ def __get_concurrency_maps(
             task_map[(dag_id, task_id)] = count
         return dag_map, task_map
 
+    def _remove_tis_with_missing_dag(self, task_instances, session=None):
+        """
+        Fail task instances and the corresponding DagRun if the dag can't be found in
+        the dags folder but exists in SerializedDag table.
+        Return task instances that exists in SerializedDag table as well as dags folder.
```

Comment:
```suggestion

        Return task instances that exists in SerializedDag table as well as dags folder.
```

Reply:
Thanks for the reviews. I will work out a similar thing like `SchedulerJob._clean_tis_without_dagrun` and see how it works